### PR TITLE
Feature specs to test most used request show features

### PR DESCRIPTION
**Pull request by danidoni**
This PR adds a feature spec to test the main features of the request show page under the beta flag.
- [x] Multiple actions
- [x] Staging requetst

### CODE REVIEW

File: src/api/spec/features/webui/requests/submissions_spec.rb [hunk 1]
```diff
@@ -103,5 +103,56 @@
         expect(page).to have_text('In state new')
       end
     end
+
+    describe 'submit several packages at once' do
+      it 'shows the beta version of the requests page' do
+        User.session = create(:admin_user)
```
Comment (hennevogel) [inline src/api/spec/features/webui/requests/submissions_spec.rb:109]: `login`

File: src/api/spec/features/webui/requests/submissions_spec.rb [hunk 2]
```diff
@@ -103,5 +103,56 @@
...
+        factory = create(:project, name: 'openSUSE:Factory')
```
Comment (hennevogel) [inline src/api/spec/features/webui/requests/submissions_spec.rb:110]: `let`

File: src/api/spec/features/webui/requests/submissions_spec.rb [hunk 3]
```diff
@@ -103,5 +103,56 @@
...
+        sworkflow = create(:staging_workflow, project: factory)
+
+        Flipper.enable(:request_show_redesign, receiver)
```
Comment (hennevogel) [inline src/api/spec/features/webui/requests/submissions_spec.rb:113]: If this is only for the beta, the `describe` block should mention that IMHO

File: src/api/spec/features/webui/requests/submissions_spec.rb [hunk 4]
```diff
@@ -103,5 +103,56 @@
...
+        login receiver
+
+        create(:relationship, project: factory, user: receiver, role: Role.hashed['reviewer'])
```
Comment (hennevogel) [inline src/api/spec/features/webui/requests/submissions_spec.rb:116]: `let!`

File: src/api/spec/features/webui/requests/submissions_spec.rb [hunk 5]
```diff
@@ -103,5 +103,56 @@
...
+
+        # Create request to submit new files to the target package A
+        bs_request = create(
```
Comment (hennevogel) [inline src/api/spec/features/webui/requests/submissions_spec.rb:119]: `let`

File: src/api/spec/features/webui/requests/submissions_spec.rb [hunk 6]
```diff
@@ -103,5 +103,56 @@
...
+          creator: receiver,
+          target_project: factory,
+          target_package: target_package,
+          source_project: source_project,
+          source_package: source_package
+        )
+        bs_request.staging_project = sworkflow.staging_projects.first
+
+        # Create more actions to submit new files from different packages to package_b
+        ('b'..'z').each_with_index do |char, index|
```
Comment (hennevogel) [inline src/api/spec/features/webui/requests/submissions_spec.rb:130]: What's the significance of doing this 25 times?

File: src/api/spec/features/webui/requests/submissions_spec.rb [hunk 7]
```diff
@@ -103,5 +103,56 @@
...
+
+          action_attributes = {
+            source_package: source_package,
+            source_project: source_project,
+            target_project: factory,
+            target_package: target_package_b
+          }
+          bs_req_action = build(:bs_request_action, action_attributes.merge(type: 'submit', bs_request: bs_request))
+          bs_req_action.save! if bs_req_action.valid?
+        end
```
Comment (hennevogel) [inline src/api/spec/features/webui/requests/submissions_spec.rb:148]: All of this is a factory.

File: src/api/spec/support/vcr.rb [hunk 8]
```diff
@@ -4,7 +4,7 @@
 VCR.configure do |config|
   config.cassette_library_dir = 'spec/cassettes'
   config.hook_into :webmock
-  config.default_cassette_options = { record: :once }
+  config.default_cassette_options = { record: :new_episodes }
```
Comment (hennevogel) [inline src/api/spec/support/vcr.rb:7]: huh?

**Review by hennevogel — CHANGES_REQUESTED** [review]

File: src/api/spec/support/vcr.rb (hunk 8 above)
Comment (danidoni) [reply src/api/spec/support/vcr.rb:7]: This is me being sloppy. I'll go and change this and all the above

File: src/api/spec/features/webui/requests/submissions_spec.rb [hunk 9]
```diff
@@ -103,5 +103,56 @@
         expect(page).to have_text('In state new')
       end
     end
+
+    describe 'when under the beta program' do
+      before { Flipper.enable(:request_show_redesign, receiver) }
```
Comment (saraycp) [inline src/api/spec/features/webui/requests/submissions_spec.rb:108]: As an alternative you can:

```suggestion
    describe 'when under the beta program', beta: true do
```

To make this work, you should enable the feature by adding
`Flipper.enable(:request_show_redesign)`
to [spec/support/beta.rb](https://github.com/openSUSE/open-build-service/blob/e90a6f1a1c0ae800b140b3b4881523f1f7a6b12a/src/api/spec/support/beta.rb#L4) what we haven't done with `request_show_redesign` yet.

File: src/api/spec/features/webui/requests/submissions_spec.rb [hunk 10]
```diff
@@ -103,5 +103,56 @@
...
+                 target_project: factory,
+                 target_package: target_package_b)
+        end
+
+        before do
+          factory
+          staging_workflow
+          # Create another action to submit new files from different packages to package_b
+          another_bs_request_action
+        end
```
Comment (saraycp) [inline src/api/spec/features/webui/requests/submissions_spec.rb:145]: I'd remove this before block and would use `let!` above to force to create these three values.

File: src/api/spec/features/webui/requests/submissions_spec.rb [hunk 11]
```diff
@@ -103,5 +103,56 @@
...
+
+        it 'shows the beta version of the requests page' do
+          login receiver
+          visit request_show_path(bs_request.number)
+
+          expect(page).to have_text("Request #{bs_request.number}")
```
Comment (saraycp) [inline src/api/spec/features/webui/requests/submissions_spec.rb:151]: This text is also present in the non-beta view, so I'd remove this expect.

A text that is beta exclusively and it seems it's going to stay is: `Comments & Request History`

**Review by saraycp — CHANGES_REQUESTED** [review]

File: src/api/spec/features/webui/requests/submissions_spec.rb (hunk 11 above)
Comment (danidoni) [reply src/api/spec/features/webui/requests/submissions_spec.rb:151]: Okay, I'll remove this expectation. But I won't add the other, I'll just rely on the existing expectations that are not present in the non beta request show page.

File: src/api/spec/features/webui/requests/submissions_spec.rb [hunk 12]
```diff
@@ -105,9 +105,7 @@
     end
 
     describe 'when under the beta program' do
-      before { Flipper.enable(:request_show_redesign, receiver) }
-
-      describe 'submit several packages at once against a factory staging project' do
+      describe 'submit several packages at once against a factory staging project', beta: true do
```
Comment (saraycp) [inline src/api/spec/features/webui/requests/submissions_spec.rb:108]: I think it makes more sense to add `beta: true` to the previous describe, which sets the "context" "_when under the beta program_".

Comment (danidoni) [reply src/api/spec/features/webui/requests/submissions_spec.rb:108]: So true, changing

**Review by saraycp — APPROVED** [review]
LGTM now. Could you please squash the commits before merging?

**Review by eduardoj — APPROVED** [review]

**Pull request merged**